src/frontend/src/repositories/anchorMetadata.test.ts

```diff
@@ -0,0 +1,223 @@
+import { MetadataMapV2 } from "$generated/internet_identity_types";
+import {
+  AnchorMetadata,
+  AnchorMetadataRepository,
+  RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+} from "./anchorMetadata";
+
+const recoveryPageShownTimestampMillis = 1234567890;
+const mockRawMetadata: MetadataMapV2 = [
+  [
+    RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+    { String: String(recoveryPageShownTimestampMillis) },
+  ],
+];
+const mockAnchorMetadata: AnchorMetadata = {
+  recoveryPageShownTimestampMillis,
+};
+
+// Used to await that the getter has resolved.
+let getterResponse: MetadataMapV2 | null | undefined = null;
+const getterMockSuccess = vi.fn().mockImplementation(async () => {
+  // The `await` is necessary to make sure that the `getterResponse` is set before the test continues.
+  getterResponse = await mockRawMetadata;
+  return mockRawMetadata;
+});
+const getterMockError = vi.fn().mockImplementation(async () => {
+  // The `await` is necessary to make sure that the `getterResponse` is set before the test continues.
+  getterResponse = await null;
+  throw new Error("test error");
+});
+const setterMockSuccess = vi.fn();
+const setterMockError = vi.fn().mockRejectedValue("test error");
+
+beforeEach(() => {
+  getterResponse = undefined;
+  vi.clearAllMocks();
+  vi.spyOn(console, "warn").mockImplementation(() => {});
+});
+
+test("AnchorMetadataRepository loads data on init in the background", async () => {
+  const instance = AnchorMetadataRepository.init({
+    getter: getterMockSuccess,
+    setter: setterMockSuccess,
+  });
+
+  await vi.waitFor(() => expect(getterResponse).toEqual(mockRawMetadata));
+
+  expect(getterMockSuccess).toHaveBeenCalledTimes(1);
+  expect(await instance.getMetadata()).toEqual(mockAnchorMetadata);
+});
+
+test("AnchorMetadataRepository loads data again on init in the background", async () => {
```

Comment:
See comment below, but I think the metadata should not be loaded _again_.

Reply:
Done.